### PR TITLE
Added TypeMatch to be used with GetterOfProducts

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

This allows GetterOfProducts to completely emulate consumesMany.

#### PR validation:

The new unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/interface/TypeMatch.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+\author C Jones, created 14 March, 2023
+
+*/
+
+#include "DataFormats/Provenance/interface/BranchDescription.h"
+
+namespace edm {
+  class TypeMatch {
+  public:
+    bool operator()(edm::BranchDescription const& branchDescription) const { return not branchDescription.isAlias(); }
```
Comment (makortel) [inline FWCore/Framework/interface/TypeMatch.h:20]: I think this should be
```suggestion
    bool operator()(edm::BranchDescription const& branchDescription) const { return not branchDescription.isAnyAlias(); }
```
to cover the aliases made by `SwitchProducer` as well.

**Comment by smuzaffar** [comment]
@Dr15Jones , I can not  find `FWCore/Framework/test/test_get_by_type_cfg.py` in repository. If it is a new file then may be you forgot to add it?

File: FWCore/Framework/interface/TypeMatch.h (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/Framework/interface/TypeMatch.h]: done

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by makortel** [comment]
+core

**Comment by perrotta** [comment]
+1

**Pull request merged**